nix/default.nix

```diff
@@ -1,7 +1,11 @@
 # This is the main entrypoint for the magic-folder derivation.
 { pkgs ? import <nixpkgs> { } }:
 let
-  pkgs' = pkgs.extend (import ./overlays.nix);
+  tahoe-repo = (import ./tahoe-lafs-repo.nix);
+  pkgs' = pkgs.appendOverlays [
```

Comment:
I didn't actually know the difference between pkgs.extend and pkgs.appendOverlays.  I didn't find any documentation about them, either.  The implementation does not suggest there is much difference at all:

```
    # Extend the package set with zero or more overlays. This preserves                                                                                                                                            
    # preexisting overlays. Prefer to initialize with the right overlays                                                                                                                                           
    # in one go when calling Nixpkgs, for performance and simplicity.                                                                                                                                              
    appendOverlays = extraOverlays:
      if extraOverlays == []
      then self
      else import ./stage.nix (args // { overlays = args.overlays ++ extraOverlays; });

    # Extend the package set with a single overlay. This preserves                                                                                                                                                 
    # preexisting overlays. Prefer to initialize with the right overlays                                                                                                                                           
    # in one go when calling Nixpkgs, for performance and simplicity.                                                                                                                                              
    # Prefer appendOverlays if used repeatedly.                                                                                                                                                                    
    extend = f: self.appendOverlays [f];
```

My experience with overlays so far has been that they're very fragile and extremely difficult to compose so I would be completely unsurprised that despite this seeming equivalence, there is still some reason `appendOverlays` is needed here rather than `extend`.

Maybe it has to do with the order?  Here we pass [tahoe's, ours] but if use tahoe's default.nix then I think it ends up as [ours, tahoe's].  Any dependencies between what we're overriding and what tahoe is overriding could end up with different derivations as a result of this different order...

Reply:
The difference is that `appendOverlays` takes a list, whereas `extend` takes a single overlay.

> Maybe it has to do with the order?

Yeah, there are two things:
- The changes to our overlay to make it compose well aren't in the tahoe overlay, so we need to apply our after the tahoe overlay.
- Some the packages we depend on directly are ones that are modified in the tahoe overlay, so we want to define our packages in the context of the tahoe overlay (I don't *think* this depends on the order of the overlays, but it depends on tahoe's overlay being applied at the top-level, rather than just for the tahoe-lafs pacakge).